plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java

```diff
@@ -250,10 +258,14 @@ public CNode describe(T instance, ConfigurationContext context) throws Exception
             Object value = a.getValue(instance);
             if (value != null) {
                 Object converted = Stapler.CONVERT_UTILS.convert(value, a.getType());
-                if (converted instanceof List || !a.isMultiple()) {
+                if (converted instanceof Collection || !a.isMultiple()) {
                     args[i] = converted;
                 } else {
-                    args[i] = Collections.singletonList(converted);
+                    if (Set.class.isAssignableFrom(p.getType())) {
+                        args[i] = Collections.singleton(converted);
+                    } else {
+                        args[i] = Collections.singletonList(converted);
```

Comment:
Non-Set and non-List Collection types will end up here after the change. Should it actually be inverted to use  `Collections.singleton` for all non-List types?

Reply:
The old behavior was to always return a list. I can invert the condition if you like.